explorer/syntax/parser.ypp

```diff
@@ -121,6 +121,8 @@
 %type <std::vector<Nonnull<Declaration*>>> interface_body
 %type <std::vector<Nonnull<Declaration*>>> impl_body
 %type <Nonnull<Statement*>> statement
+%type <Nonnull<Statement*>> assignment_statement
+%type <AssignOperator> assignment_operator;
```

Comment:
Why assign versus assignment?

Reply:
`AssignOperator`  and `class Assign` were chosen to mirror the `AssignBlah` interfaces (eg, `AddAssign` corresponds to `AssignOperator::Add`). `assignment_statement` was chosen to match the name of the statement in the design.

I'm not at all tied to this naming. If you'd prefer `AssignmentOperator` and `class Assignment` for the statement, or `assign_statement` for the grammar production, I'd be OK with either of those changes.

Reply:
But `class Assign` is a statement, not an operator... If you want to be consistent with the design, shouldn't it be `class Assignment`, `assignment_statement`, `AssignOperator`, and `assign_operator`?

Note, I actually assumed `assign` was being used for brevity versus `assignment`. I'm fine with most of the matrix of options here, I'm just objecting specifically to `AssignOperator assignment_operator` because it seems like the same naming scheme should be used for `assign` versus `assignment` in both of those. They're talking about the same thing.

PS: as I'm looking at this again -- I think you should remove the semicolon after `assignment_operator;` here.

Reply:
I was actually considering whether `AssignOperator` should be `Assign::Operator` if you want my attitude on this...

Reply:
Let's be consistent within explorer. Switched to just "assign" everywhere.

Reply:
Suggestion for moving `AssignOperator` inside `Assign` noted; may send a follow-up to do that here and for `Operator`.